src/ray/rpc/server_call.h

```diff
@@ -223,18 +266,24 @@ class ServerCallImpl : public ServerCall {
       // a new request comes in.
       factory.CreateCall();
     }
-    (service_handler_.*handle_request_function_)(
-        std::move(request_),
-        reply_,
-        [this](
-            Status status, std::function<void()> success, std::function<void()> failure) {
-          // These two callbacks must be set before `SendReply`, because `SendReply`
-          // is async and this `ServerCall` might be deleted right after `SendReply`.
-          send_reply_success_callback_ = std::move(success);
-          send_reply_failure_callback_ = std::move(failure);
-          boost::asio::post(GetServerCallExecutor(),
-                            [this, status]() { SendReply(status); });
-        });
+    if (!auth_success) {
+      boost::asio::post(GetServerCallExecutor(),
+                        [this]() { SendReply(Status::AuthError("WrongClusterToken")); });
```

Comment:
Maybe add more details about what could be the cause. Somehting like a raylet accessing the GCS which doesn't belong to that cluster?
It should be useful for the users to debug.